pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexCRUDViolationRule.java

```diff
@@ -506,7 +521,9 @@ private void validateCRUDCheckPresent(final AbstractApexNode<?> node, final Obje
             final String typeCheck) {
         if (!typeToDMLOperationMapping.containsKey(typeCheck)) {
             if (!isProperESAPICheckForDML(typeCheck, crudMethod)) {
-                addViolation(data, node);
+                if (!isWithSecurityEnforced(node)) {
```

Comment:
Please combine the 3 if statements to avoid nesting:

```
if (!typeToDMLOperationMapping.containsKey(typeCheck)
    && !isProperESAPICheckForDML(typeCheck, crudMethod)
    && !isWithSecurityEnforced(node)) {
...
```